daffodil-core/src/main/scala/org/apache/daffodil/core/grammar/ElementBaseGrammarMixin.scala

```diff
@@ -1392,6 +1392,7 @@ trait ElementBaseGrammarMixin
           notYetImplemented("lengthKind='endOfParent' for complex type")
         case LengthKind.EndOfParent =>
           notYetImplemented("lengthKind='endOfParent' for simple type")
+        case _ => Assert.impossibleCase("default case should not be reached")
```

Comment:
What case does scala think is not matched, seems like that might be a bug?

Reply:
I think it said Delimited and Implicit

Reply:
If it's only a couple of cases, can we explicitly add them as impossible cases? This match case is complicated enough that it might be helpful to self document exactly what cases aren't supported here.